repo_remote.go

```diff
@@ -146,3 +146,181 @@ func RepoRemoveRemote(repoPath, name string, opts ...RemoveRemoteOptions) error
 func (r *Repository) RemoveRemote(name string, opts ...RemoveRemoteOptions) error {
 	return RepoRemoveRemote(r.path, name, opts...)
 }
+
+// RemotesListOptions contains arguments for listing remotes of the repository.
+// Docs: https://git-scm.com/docs/git-remote#_commands
+type RemotesListOptions struct {
+	// The timeout duration before giving up for each shell command execution.
+	// The default timeout duration will be used when not supplied.
+	Timeout time.Duration
+}
+
+// RepoRemotesList lists remotes of the repository in given path.
+func RepoRemotesList(repoPath string, opts ...RemotesListOptions) ([]string, error) {
```

Comment:
If we're following how we use git bin, this should be like this? (if so, same for other methods and option structs)
```suggestion
func RepoRemotes(repoPath string, opts ...RemotesListOptions) ([]string, error) {
```

Reply:
Renamed it, though kept `RemotesOptions` might get confusing?